### PR TITLE
Request Body examples should respect media-type

### LINKED ISSUE #6475: requestBody examples do not respect media-type
### Q&A (please complete the following information)
OS: [e.g. macOS] Windows 10 Pro
Browser: [e.g. chrome, safari] chrome
Version: [e.g. 22] Version 85.0.4183.102 (Official Build) (64-bit)
Method of installation: [e.g. npm, dist assets] raw code
Swagger-UI version: [e.g. 3.10.0] master
Swagger/OpenAPI version: [e.g. Swagger 2.0, OpenAPI 3.0] 3.0.1
### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.1
info:
 title: Example Swagger
 version: '1.0'
servers:
 - url: /api/v1
paths:
 /xmlTest:
   post:
     summary: sample issues
     operationId: registerSubscription
     parameters: []
     requestBody:
       description: Simple Test
       content:
           application/xml:
              schema:
               $ref: "#/components/schemas/Test"
              example: 
                x: should be xml
     responses:
       '200':
         description: Simple example
         content: {}
components:
 schemas:
   Test:
     type: object
     xml:
       name: root
     properties:
       x:
         type: string
       other:
         type: string
         format: email
     example: 
       x: what the f
```

### Describe the bug you're encountering

When providing examples via `examples` or `example` keyword for requestBody content it will not be rendered according to content-type.

### To reproduce...
Open yaml at editor and try to view xml requestBody sample. You can delete the `example` part and it will render it in xml.

### Expected behavior
Examples provided via `examples` or `example` for requestBody definition should be rendered respecting the content-type.
With `example` requestBody:
image
Without:
image

### Additional context or thoughts
Same as for Responses #6442

### PR DESCRIPTION
**Pull request by mathis-m**
### Description
Request Body samples do not respect the media type selected.
It just stringifies the json representation of the sample provided via `examples` or `example` keyword.
It should use the `getSampleSchema` method, which does take care of the content-type // media-type.

### How Has This Been Tested?

Fixes #6475 

### Screenshots (if appropriate):
see #6475 

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mathis-m** [comment]
@tim-lai its been a while, could you have a look at this PR and pull it into mainline. It fixes the sample generation for request body related to #6470 .

File: src/core/plugins/oas3/components/request-body.jsx [hunk 1]
```diff
@@ -212,6 +209,12 @@ const RequestBody = ({
     </div>
   }
 
+  let sampleRequestBody = getDefaultRequestBodyValue(
```
Comment (mathis-m) [inline src/core/plugins/oas3/components/request-body.jsx:212]: could be const

**Comment by tim-lai** [comment]
@mathis-m Catching up on holiday backlog. This PR is now merged. Thanks for the fix!

**Comment by mathis-m** [comment]
@tim-lai nice :) best wishes for the new year!

**Pull request merged**